### PR TITLE
[tracing] Improve tracing for TS components

### PR DESCRIPTION
**Pull request by geropl**
## Description
Mainly contains two changes (+ minor cleanup)
 - improve API level tracing: https://github.com/gitpod-io/gitpod/commit/7e28d67e433fc45e24eeeb915643d0f4f0cf57c4
 - use less `span.log(...)` calls:
    - https://github.com/gitpod-io/gitpod/commit/2105d4775eab924347c9b5f5472c17c129828cf6
    - https://github.com/gitpod-io/gitpod/commit/78ce11bbd10cc540f4501cbd6fd3acc1d0217e89

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Documentation

/cc @csweichel

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/ee/src/prebuilds/prebuilt-status-maintainer.ts
Comment (mads-hartmann) [inline components/server/ee/src/prebuilds/prebuilt-status-maintainer.ts:180]: I think it's worth trying to namespace these. We don't have a convention established for how to generically namespace attributes for custom spans, but how just going namespacing them under the name of the span? 

That would make it `doUpdate.update` and `doUpdate.found`.

I think that gives us fairly readable queries. E.g. to find attributes related to the span you care about you'd type `doUpdate.` and use code completion to explore the attributes. Queries would then textually looks something like:

```
COUNT
WHERE name = doUpgrade
GROUP BY doUpgrade.update, doUpgrade.found
```

File: components/gitpod-protocol/src/util/tracing.ts
Comment (mads-hartmann) [inline components/gitpod-protocol/src/util/tracing.ts:35]: This method seems superfluous to me. In the two paces we use it I think it would be nicer just to 

```ts
const span = TraceContext.startSpan(...)
// and then use { span } when it needs to be passed to something that expects a ctx instead of a span
```

File: components/server/ee/src/prebuilds/prebuilt-status-maintainer.ts
Comment (mads-hartmann) [inline components/server/ee/src/prebuilds/prebuilt-status-maintainer.ts:220]: If we remove the childContext method we could just `span.finish()` here

File: components/gitpod-protocol/src/util/tracing.ts
Comment (mads-hartmann) [inline components/gitpod-protocol/src/util/tracing.ts:40]: For methods that only need access to the span I don't think we need to pass in the full context. But that would mean also changing addNestedTags so out of scope of this PR I think

**Review by mads-hartmann — COMMENTED** [review]
Thanks! I left a few clarifying comments and suggestions.

File: components/gitpod-protocol/src/util/tracing.ts
Comment (geropl) [reply components/gitpod-protocol/src/util/tracing.ts:40]: We always pass `TraceContext`, and I feel we should keep the pattern. Especially as it's just a `{ }` wrapper.

File: components/server/ee/src/prebuilds/prebuilt-status-maintainer.ts
Comment (geropl) [reply components/server/ee/src/prebuilds/prebuilt-status-maintainer.ts:180]: Good point, will do.

File: components/gitpod-protocol/src/util/tracing.ts
Comment (geropl) [reply components/gitpod-protocol/src/util/tracing.ts:35]: IMO this is out-of scope here but:
 - either we keep the current style (`TraceContext`), then it's trade-off between `{ span }` on every call vs. `ctx`
 - or we switch to `span` directly
I see no pressure atm to switch styles.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7170#issuecomment-990969575" title="Approved">geropl</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7170#issuecomment-990969575" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [geropl]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]
- ~~[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by mads-hartmann — APPROVED** [review]
Thanks!

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 0cd67b013f7470a8d2b22462854c5467231b6eb3</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7170#issuecomment-990969575" title="Approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7170#pullrequestreview-828887600" title="Approved">mads-hartmann</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7170#issuecomment-990969575" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]
- ~~[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)~~ [geropl]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]
- ~~[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
